.github/workflows/slow-tests.yml

```
@@ -0,0 +1,89 @@
+# This workflow is triggered for every push to master and runs all bazel tests
+# that take too long to run on PRs in ci-main/bazel-test-all.
+#
+# The following chart shows the 90th percentile of the duration per bazel test in ci-main/bazel-test-all:
+# https://superset.idx.dfinity.network/explore/?slice_id=48
+# If a test takes longer than 5 minutes it has to be tagged with "slow_test"
+# to skip running it in ci-main/bazel-test-all and run it here instead.
+
+name: Slow Tests
```

```suggestion
name: CI Main Long
```
Also rename the file to `ci-main-long.yml`.

---

Ok yes that makes sense so that it shows these are tests originally from ci-main but take too long. But then I think I should also rename the job and bazel label accordingly. Will do.

---

On a second thought, shouldn't we just use the same workflow and use include `slow_test` in the tag selection on push event?

---

I thought about that but decided against it because with a different job it's easier to filter in Superset. And frankly also easier to explain to people.

---

It's as easy - it's just a different event, not `pull_request` but `push`. Better that then overhead with additional workflow.

---

But we will have to update all charts in Superset to only show bazel-test-all data from pull_requests and not for push events.

---

> But we will have to update all charts in Superset to only show bazel-test-all data from pull_requests and not for push events.

No, we already do that because of `merge_group` event. See [here](https://github.com/dfinity/ic/blob/master/.github/workflows-source/ci-main.yml#L117) why.

---

What I mean is that charts like [Bazel Test All / Critical Path / From Last Week Onwards](https://superset.idx.dfinity.network/explore/?form_data_key=fWO_2gF0vx5IsQp6UMkLBJeP-Qc3cORNADUeXC2WQtaAgRicGUfRCyrC9ITuR0fz&dashboard_page_id=a96-xRXbl&slice_id=35&native_filters_key=WZi8efszqAxY49KiT2uiVuqcDHRXzAKSaguZaWCf0wHW3bwX5s9GAl4ttQjqNzj_) show all events from bazel-test-all regardless of whether it ran on PRs or on master. At the moment that's not a big issue but when bazel-test-all starts running long_tests on master that data will show up in those charts defeating the point of those charts since they should only show statistics on PRs.

An additional issue is that folks have also [asked](https://dfinity.slack.com/archives/CH4CADCJX/p1729056329710249?thread_ts=1729024031.315719&cid=CH4CADCJX) how to run these tests slow_tests manually. The problem with running this as part of `CI Main` is that when you trigger `CI Main` via the `workflow_dispatch` trigger the data will also show up in the Superset charts.

We might be able to add additional filters to the charts to only show PR events but I think it's simpler if different tests run in a different job and workflow.

---

So we could add a filter to those charts like:
```
head_branch <> `master`
```
but that won't work for manually triggered CI Main runs via `workflow_dispatch`.

---

If these tests need to be run on PR, labels can be used and updating the charts (so that on push events are filtered out) is straightforward. If you see that this is better for developers, go ahead, but I'm not convinced so far.

---

> updating the charts (so that on push events are filtered out) is straightforward

I see. I guess we can update most datasets by additionally joining them with `workflow_runs` and filter on `is_pull_request` / `event_type`.

---

Ok, slow_tests now run as part of CI Main unless on PRs.